src/rtk/features/overviewOrg/overviewOrgSlice.ts

```diff
@@ -17,69 +23,94 @@ export interface OverviewOrg {
   apiList: ApiListItem[];
 }
 
-export interface InitialState {
+interface DelegationDTO {
+  coveredByName: string;
+  offeredByName: string;
+  offeredByOrganizationNumber: string;
+  coveredByOrganizationNumber: string;
+  resourceId: string;
+  resourceTitle: languageDto;
+}
+
+export interface SliceState {
   loading: boolean;
   overviewOrgs: OverviewOrg[];
   error: string;
+  layout: LayoutState;
 }
 
-const initialState: InitialState = {
-  loading: false,
-  overviewOrgs: [
-    {
-      id: '1',
-      orgName: 'Netcompany',
-      orgNr: '123456789',
-      isAllSoftDeleted: false,
-      apiList: [
-        {
-          id: '1',
-          apiName: 'Delegert API A',
-          isSoftDelete: false,
-          owner: 'Brønnøysundregisterene',
-          description:
-            'kan du registrere og endre opplysninger på bedrift, finne bedriftsinformasjon og kunngjøringer, sjekke heftelser i bil og stoppe telefonsalg.',
-        },
-        {
-          id: '2',
-          apiName: 'Delegert API B',
-          isSoftDelete: false,
-          owner: 'Accenture',
-          description:
-            'API for forvaltningsorgan og kompetansesenter som skal styrke kommunenes, sektormyndighetenes og andre samarbeidspartneres kompetanse på integrering og',
-        },
-      ],
-    },
-    {
-      id: '2',
-      orgName: 'Tieto',
-      orgNr: '950124321',
-      isAllSoftDeleted: false,
-      apiList: [
-        {
-          id: '1',
-          apiName: 'Delegert API A',
-          isSoftDelete: false,
-          owner: 'Avanade',
-          description:
-            'kan du registrere og endre opplysninger på bedrift, finne bedriftsinformasjon og kunngjøringer, sjekke heftelser i bil og stoppe telefonsalg.',
-        },
-        {
-          id: '2',
-          apiName: 'Delegert API B',
-          isSoftDelete: false,
-          owner: 'Accenture',
-          description:
-            'Accenture er et forvaltningsorgan og kompetansesenter som skal styrke kommunenes, sektormyndighetenes og andre samarbeidspartneres kompetanse på integrering og',
-        },
-      ],
-    },
-  ],
+interface languageDto {
+  en: string;
+  nb: string;
+  nn: string;
+}
+
+const initialState: SliceState = {
+  loading: true,
+  overviewOrgs: [],
   error: '',
+  layout: LayoutState.Given,
+};
+
+const mapToOverviewOrgList = (delegationArray: DelegationDTO[], layout: LayoutState) => {
```

Comment:
I'm not sure if this method should be split in two or not. Will there be a lot fo duplicate code if we do so?

Reply:
Probably, yeah. Everything but the 2nd switch case would have to be duplicate code I would assume 🤔

Reply:
Then we should probably keep this method like it already is 😊